makefile

```diff
@@ -1,7 +1,9 @@
 install:
-	pip install pip==23.3.1
-	pip install -e '.[dev,docs]'
+	pip install uv==0.2.34
+	uv pip install -r requirements.txt --editable .
```

Comment:
I would prefer this to be `uv pip sync requirements.txt` but this causes [CI to fail](https://app.circleci.com/pipelines/github/octoenergy/xocto/637/workflows/fc322f08-a5bb-4495-a086-e9dff0aeb60d/jobs/619) with `importlib.metadata.PackageNotFoundError: xocto`